src/app/+search-page/search-results/search-results.component.spec.ts

```diff
@@ -9,32 +10,50 @@ import { SearchResultsComponent } from './search-results.component';
 describe('SearchResultsComponent', () => {
   let comp: SearchResultsComponent;
   let fixture: ComponentFixture<SearchResultsComponent>;
-  let heading: DebugElement;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [TranslateModule.forRoot()],
+      imports: [TranslateModule.forRoot(), NoopAnimationsModule],
       declarations: [SearchResultsComponent],
       schemas: [NO_ERRORS_SCHEMA]
     }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(SearchResultsComponent);
-    comp = fixture.componentInstance; // SearchFormComponent test instance
-    heading = fixture.debugElement.query(By.css('heading'));
+    comp = fixture.componentInstance; // SearchResultsComponent test instance
   });
 
-  it('should display heading when results are not empty', fakeAsync(() => {
-    (comp as any).searchResults = 'test';
-    (comp as any).searchConfig = {pagination: ''};
+  it('should display results when results are not empty', () => {
+    (comp as any).searchResults = { hasSucceeded: true, isLoading: false, payload: { page: { length: 2 } } };
+    (comp as any).searchConfig = {};
     fixture.detectChanges();
-    tick();
-    expect(heading).toBeDefined();
-  }));
+    expect(fixture.debugElement.query(By.css('ds-viewable-collection'))).not.toBeNull();
+  });
+
+  it('should not display link when results are not empty', () => {
+    (comp as any).searchResults = { hasSucceeded: true, isLoading: false, payload: { page: { length: 2 } } };
+    (comp as any).searchConfig = {};
+    fixture.detectChanges();
+    expect(fixture.debugElement.query(By.css('a'))).toBeNull();
+  });
+
+  it('should display error message if error is != 400', () => {
+    (comp as any).searchResults = { hasFailed: true, error: { statusCode: 500 } };
+    fixture.detectChanges();
+    expect(fixture.debugElement.query(By.css('ds-error'))).not.toBeNull();
+  });
+
+  it('should display link with new search where query is quoted if search return a error 400', () => {
+    (comp as any).searchResults = { hasFailed: true, error: { statusCode: 400 } };
+    fixture.detectChanges();
+    expect(fixture.debugElement.query(By.css('a'))).not.toBeNull();
```

Comment:
This doesn't test what the link does, only that there is a link. 

Ideally you'd test: 
- that the link you navigate to contains the same query but in quotes
- that any other params are still there, and haven't been modified.

[This section of the angular.io testing guide](https://angular.io/guide/testing#components-with-routerlink) can help with that

Reply:
I solved the first part of your suggestion. But I am not sure about how to test

> that any other params are still there, and haven't been modified.

I tried providing a mocked ActivatedRoute, with some params set, to TestBed. But I am not sure how I can test how the resulting link will look like as the ComponentFixture contains only the router link before merging the parameters. 

And I am not sure if this test is really necessary as it is just testing if Angulars queryParamsHandling is working as expected. Or am I missing something here?